skimage/future/detect_obj/cascade.pyx

```diff
@@ -0,0 +1,832 @@
+# and the average can be easily computed in a constant time.
+cdef struct DetectionsCluster:
+
+    int r_sum
+    int c_sum
+    int width_sum
+    int height_sum
+    int count
+
+
+# Struct for storing a single detection.
+cdef struct Detection:
+
+    int r
+    int c
+    int width
+    int height
+
+
+# Struct for storing multi-block binary pattern position.
+cdef struct MBLBP:
+
+    Py_ssize_t r
+    Py_ssize_t c
+    Py_ssize_t width
+    Py_ssize_t height
+
+
+# Struct for storing a stump of classifying cascade. It has the index to the
+# look-up table which is stored in Cascade class. Depending on the value of
+# the feature after its evaluation `left` or `right` value is returned which
+# is used by Cascade classifier to predict whether or not the
+# object is detected.
+cdef struct MBLBPStump:
+
+    Py_ssize_t feature_id
+    Py_ssize_t lut_idx
+    float left
+    float right
+
+
+# Struct for storing a stage of classifier which itself consists of stumps.
+# It has the index that maps to the starting stump and amount of stumps.
+# In each stage all the stumps are evaluated and their output values( `left`
+# or `right` depending on the input) are summed up and compared to the
+# threshold. If the value is higher than the threshold, the stage is passed
+# and Cascade classifier goes to the next one. If all the stages are passed,
+# the object is predicted to be present in the input image patch.
+cdef struct Stage:
+
+    Py_ssize_t first_idx
+    Py_ssize_t amount
+    float threshold
+
+
+cdef vector[Detection] _group_detections(vector[Detection] detections,
+                                         float intersection_score_threshold=0.5,
+                                         int min_neighbour_amount=4):
+    """Groups similar detection into single detection and eliminates weak
+    detections that have small amount of intersecting detection.
+
+    The function assumes that true detections are supposed to have a certain
+    amount of intersecting detections and the false detections are supposed to
+    have small amount. Based on this approach, the final detections are
+    computed. The detections that are approved and gathered into one cluster
+    are averaged and this value is returned.
+
+    Parameters
+    ----------
+    detections : vector[Detection]
+        A cluster of detections.
+    min_neighbour_amount : int
+        Minimum amount of intersecting detections in order for detection
+        to be approved by the function.
+    intersection_score_threshold : float
+        The minimum value of value of ratio
+        (intersection area) / (small rectangle ratio) in order to merge
+        two rectangles into one cluster.
+
+    Returns
+    -------
+    output : vector[Detection]
+        The grouped detections.
+    """
+
+    cdef:
+        Detection mean_detection
+        vector[DetectionsCluster] clusters
+        vector[int] clusters_scores
+        Py_ssize_t clusters_amount
```

Comment:
clusters_amount is a strange variable naming--what does it signify?  Perhaps nr_of_clusters or cluster_overlap_nr etc. depending on what it means.